public/app/features/explore/state/explorePane.test.ts

```diff
@@ -30,134 +21,125 @@ const testRange = {
+      query: jest.fn(),
+      name: 'someDs',
+      meta: { id: 'someDs' },
+    },
   };
-};
 
-describe('refreshExplore', () => {
-  describe('when explore is initialized', () => {
-    describe('and update datasource is set', () => {
-      it('then it should dispatch initializeExplore', async () => {
-        const { exploreId, initialState, containerWidth, eventBridge } = setup({ datasource: true });
-
-        const dispatchedActions = await thunkTester(initialState)
-          .givenThunk(refreshExplore)
-          .whenThunkIsDispatched(exploreId);
-
-        const initializeExplore = dispatchedActions.find((action) => action.type === initializeExploreAction.type);
-        const { type, payload } = initializeExplore as PayloadAction<InitializeExplorePayload>;
-
-        expect(type).toEqual(initializeExploreAction.type);
-        expect(payload.containerWidth).toEqual(containerWidth);
-        expect(payload.eventBridge).toEqual(eventBridge);
-        expect(payload.queries.length).toBe(1); // Queries have generated keys hard to expect on
-        expect(payload.range.from).toEqual(testRange.from);
-        expect(payload.range.to).toEqual(testRange.to);
-        expect(payload.range.raw.from).toEqual(testRange.raw.from);
-        expect(payload.range.raw.to).toEqual(testRange.raw.to);
-      });
-    });
-
-    describe('and update queries is set', () => {
-      it('then it should dispatch setQueriesAction', async () => {
-        const { exploreId, initialState } = setup({ queries: true });
-
-        const dispatchedActions = await thunkTester(initialState)
-          .givenThunk(refreshExplore)
-          .whenThunkIsDispatched(exploreId);
+  setDataSourceSrv({
+    getList() {
+      return Object.values(datasources).map((d) => ({ name: d.name }));
+    },
+    getInstanceSettings(name: string) {
+      return { name: 'hello' };
+    },
+    get(name?: string) {
+      return Promise.resolve(
+        name
+          ? datasources[name]
+          : {
+              testDatasource: jest.fn(),
+              init: jest.fn(),
+              name: 'default',
+            }
+      );
+    },
+  } as any);
 
-        expect(dispatchedActions[0].type).toEqual(setQueriesAction.type);
-        expect(dispatchedActions[0].payload).toEqual({ exploreId, queries: [] });
-      });
-    });
+  const store = setupStore({
+    datasourceInstance: datasources.someDs,
+    ...(state || {}),
   });
 
-  describe('when update is not initialized', () => {
-    it('then it should not dispatch any actions', async () => {
-      const exploreId = ExploreId.left;
-      const initialState = { explore: { [exploreId]: { initialized: false } } };
-
-      const dispatchedActions = await thunkTester(initialState)
-        .givenThunk(refreshExplore)
-        .whenThunkIsDispatched(exploreId);
+  return {
+    store,
+    datasources,
+  };
+}
 
-      expect(dispatchedActions).toEqual([]);
-    });
+describe('refreshExplore', () => {
```

Comment:
These tests are refactored quite a bit to not test what action are triggered but if an action actually results in expected store state.